client/client.go

```diff
@@ -32,6 +33,29 @@ import (
 	"github.com/canonical/pebble/internal/wsutil"
 )
 
+// SocketNotFoundError is the error type returned when the Pebble client fails
+// to find a unix socket at the specified path.
+type SocketNotFoundError struct {
+	// Error wrapped error.
+	Err error
+
+	// SocketPath of the non existent socket.
```

Comment:
Nit: should this say `// SocketPath is the path of the non-existent socket.` (added "is the path" and hyphen)